archiver/archiver_test.go

```diff
@@ -0,0 +1,274 @@
+package archiver
+
+import (
+	"bufio"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+
+	c "github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+
+	"github.com/rudderlabs/rudder-go-kit/bytesize"
+	"github.com/rudderlabs/rudder-go-kit/filemanager"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/jobsdb/prebackup"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+
+	arch "github.com/rudderlabs/rudder-server/services/archiver"
+	"github.com/rudderlabs/rudder-server/services/fileuploader"
+	"github.com/rudderlabs/rudder-server/testhelper"
+	"github.com/rudderlabs/rudder-server/testhelper/destination"
+)
+
+func TestJobsArchival(t *testing.T) {
+	var (
+		prefix        = "some-prefix"
+		minioResource []*destination.MINIOResource
+
+		// test data - contains jobs from 3 workspaces(1 - 1 source, 2 & 3 - 2 sources each)
+		goldenFileJobsFileName = "testdata/MultiWorkspaceBackupJobs.json.gz"
```

Comment:
[minor] Given its usage I would avoid using `goldenFile`:
* We run comparing the result of the operation with this file or the contents of this file, rather we use it as a seed for our jobs
* Change in archival functionality should require the file to change as well, which is not the case here

 ```suggestion
		seedJobsFileName = "testdata/MultiWorkspaceBackupJobs.json.gz"
```